docs/proposals/ldap-group-sync.md

```diff
@@ -0,0 +1,189 @@
+# Allowing OpenShift Groups to be Synced With Sources on an LDAP Server and Making it Extensible for Other External Sources
+
+___
+Note: in this proposal, if a reference is prefixed with "OpenShift" or is typeset in `monospace`, it refers to an OpenShift object or field. For instance, the following table enumerates what references to OpenShift resources and LDAP entries may look like:
+
+OpenShift Resource | LDAP Entry
+------------------ | ----------
+OpenShift Group    | LDAP group
+OpenShift `Group`  | group
+`Group`            |
+___
+
+Linking OpenShift `Groups` to their representations on LDAP servers allows for administrators to have an easier way to manage `Groups` inside OpenShift. This additional functionality will require no additional fields on the OpenShift `Group` object. One can indicate that an OpenShift `Group` mirrors an external representation by setting the `openshift.io/externalRecordProvider` `Annotation` in the `Group` as well as adding a `Label` such as `openshift.io/externalRecordProvider/LDAP` to the `Group`. Other information (e.g. last sync time or resource version) will be stored as `Annotations` on the OpenShift `Group`. Logistical information for the LDAP sync will be stored in an `LDAPSyncOptions` that the administrator will need to provide.
+
+### Use Cases
+* I want to sync all groups I've configured in a sync job (e.g. a specific LDAP server configuration) to OpenShift `Groups`
+* I want to sync specific OpenShift `Groups` with their `externalRecordProviders`
+
+### Data Flow
+* Determine list of LDAP groups to be synced
+* Find LDAP group members
+* Determine OpenShift `Identities` from LDAP group members
+* Determine OpenShift `Users` from OpenShift `Identities`
+* Determine OpenShift `Group` from LDAP group
+* Populate new OpenShift `Group` from resulting OpenShift `UserIdentityMappings` and old OpenShift `Group` fields
+
+The sync command will be exposed as `oadm sync <group(s)> [<group-name>] [filters]` and invoked like `oadm sync group <name>` or `oadm sync groups [--label=selector]`.
```

Comment:
I think you should move the execution examples further down.  They're distracting me because the distinction between ldap groups, openshift groups, and things I want like whitelisting and blacklisting for initialization all factor into this. It needs to be a long section with different permutations.